File: app/src/test/java/com/example/creditcardinputexercise/MainActivityTest.java
package com.example.creditcardinputexercise;

import com.google.android.material.textfield.TextInputLayout;

import org.junit.Test;
import org.w3c.dom.Text;

import static org.junit.Assert.*;

public class MainActivityTest {

    @Test
    public void field_empty() {
        String name = "";
        MainActivity activity = new MainActivity();
        boolean b = activity.field_empty(name);
        assertEquals(false, b);
    }

    @Test
    public void field_empty2() {
        String name = "234234234234";
        MainActivity activity = new MainActivity();
        boolean b = activity.field_empty(name);
        assertEquals(true, b);
    }

    @Test
    public void name_matches() {
        String name = "teretertertert";
        MainActivity activity = new MainActivity();
        boolean b = activity.name_matches(name);
        assertEquals(true, b);
    }

    @Test
    public void name_matches2() {
        String name = "t ";
        MainActivity activity = new MainActivity();
        boolean b = activity.name_matches(name);
        assertEquals(true, b);
    }

    @Test
    public void name_matches3() {
        String name = "34546456456";
        MainActivity activity = new MainActivity();
        boolean b = activity.name_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void date_matches() {
        String name = "10/20";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(true, b);
    }

    @Test
    public void date_matches1() {
        String name = "10+0020";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void date_matches2() {
        String name = "fgkjdfhgkjdfg";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void date_matches4() {
        String name = "10/99";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(true, b);
    }

    @Test
    public void date_field_length() {
        String name = "";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void date_field_length2() {
        String name = "10";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void date_field_length3() {
        String name = "10/1999";
        MainActivity activity = new MainActivity();
        boolean b = activity.date_matches(name);
        assertEquals(false, b);
    }

    @Test
    public void check_expiration() {
        String name = "08/20";
        MainActivity activity = new MainActivity();
        boolean b = activity.check_expiration(name);
        assertEquals(false, b);
    }

    @Test
    public void check_expiration2() {
        String name = "10/20";
        MainActivity activity = new MainActivity();
        boolean b = activity.check_expiration(name);
        assertEquals(true, b);
    }

    @Test
    public void check_expiration3() {
        String name = "10/19";
        MainActivity activity = new MainActivity();
        boolean b = activity.check_expiration(name);
        assertEquals(false, b);
    }
}

File: app/src/test/java/com/example/creditcardinputexercise/MainActivityTest.kt
package com.example.creditcardinputexercise

import org.junit.Assert
import org.junit.Test

class MainActivityTest {
    @Test
    fun field_empty() {
        val name = ""
        val activity = MainActivity()
        val b = activity.field_empty(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun field_empty2() {
        val name = "234234234234"
        val activity = MainActivity()
        val b = activity.field_empty(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun name_matches() {
        val name = "teretertertert"
        val activity = MainActivity()
        val b = activity.name_matches(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun name_matches2() {
        val name = "t "
        val activity = MainActivity()
        val b = activity.name_matches(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun name_matches3() {
        val name = "34546456456"
        val activity = MainActivity()
        val b = activity.name_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun date_matches() {
        val name = "10/20"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun date_matches1() {
        val name = "10+0020"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun date_matches2() {
        val name = "fgkjdfhgkjdfg"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun date_matches4() {
        val name = "10/99"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun date_field_length() {
        val name = ""
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun date_field_length2() {
        val name = "10"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun date_field_length3() {
        val name = "10/1999"
        val activity = MainActivity()
        val b = activity.date_matches(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun check_expiration() {
        val name = "08/20"
        val activity = MainActivity()
        val b = activity.check_expiration(name)
        Assert.assertEquals(false, b)
    }

    @Test
    fun check_expiration2() {
        val name = "10/20"
        val activity = MainActivity()
        val b = activity.check_expiration(name)
        Assert.assertEquals(true, b)
    }

    @Test
    fun check_expiration3() {
        val name = "10/19"
        val activity = MainActivity()
        val b = activity.check_expiration(name)
        Assert.assertEquals(false, b)
    }
}